feat: support provider states by branch/env

YOU54F opened the pull request:
Add the following endpoints to allow scoping of returned provider states, based on a named consumer branch or environment

`/pacts/provider/{provider}/provider-states/environment/{environment_name}`
`/pacts/provider/{provider}/provider-states/branch/{branch_name}`

Saup21 commented on lib/pact_broker/api.rb:
```diff
@@ -47,6 +47,8 @@ def self.build_api(application_context = PactBroker::ApplicationContext.default_
         # Provider states
 
         add ["pacts", "provider", :provider_name, "provider-states"], Api::Resources::ProviderStates, { resource_name: "provider_states" }
+        add ["pacts", "provider", :provider_name, "provider-states","branch", :branch_name], Api::Resources::ProviderStates, { resource_name: "provider_states_for_branch" }
```
```suggestion
        add ["pacts", "provider", :provider_name, "provider-states", "branch", :branch_name], Api::Resources::ProviderStates, { resource_name: "provider_states_for_branch" }
```

YOU54F replied:
sorted thx

Saup21 commented on lib/pact_broker/api.rb:
```diff
@@ -47,6 +47,8 @@ def self.build_api(application_context = PactBroker::ApplicationContext.default_
 
         add ["pacts", "provider", :provider_name, "provider-states"], Api::Resources::ProviderStates, { resource_name: "provider_states" }
+        add ["pacts", "provider", :provider_name, "provider-states","branch", :branch_name], Api::Resources::ProviderStates, { resource_name: "provider_states_for_branch" }
+        add ["pacts", "provider", :provider_name, "provider-states","environment", :environment_uuid], Api::Resources::ProviderStates, { resource_name: "provider_states_for_environment" }
```
```suggestion
        add ["pacts", "provider", :provider_name, "provider-states", "environment", :environment_uuid], Api::Resources::ProviderStates, { resource_name: "provider_states_for_environment" }
```

YOU54F replied:
sorted thx

YOU54F committed:
feat: support provider states by branch/env

    /pacts/provider/{provider}/provider-states/environment/{environment_name}
    /pacts/provider/{provider}/provider-states/branch/{branch_name}
